apollo-router/src/services/layers/persisted_queries/mod.rs

```diff
@@ -184,61 +143,98 @@ impl PersistedQueryLayer {
         }
     }
 
-    /// Handles incoming freeform GraphQL requests according to the safelisting configuration options
-    pub(crate) fn handle_freeform_graphql(
+    pub(crate) async fn supergraph_request_with_analyzed_query(
         &self,
         request: SupergraphRequest,
-        manifest_poller: Arc<PersistedQueryManifestPoller>,
     ) -> Result<SupergraphRequest, SupergraphResponse> {
-        if let Some(operation_body) = request.supergraph_request.body().query.as_ref() {
-            // TODO: replace this with proper introspection parsing
-            if self.sandbox_enabled
-                && (operation_body == SANDBOX_INTROSPECTION_QUERY_WO_DEPRECATED_ARGS
-                    || operation_body == SANDBOX_INTROSPECTION_QUERY_W_DEPRECATED_ARGS)
-            {
-                // if sandbox is enabled and the incoming operation is the introspection query sent by sandbox,
-                // allow the request to continue on.
+        let manifest_poller = match &self.manifest_poller {
+            // PQ feature entirely disabled; just pass through.
+            None => return Ok(request),
+            Some(mp) => mp,
+        };
+
+        let operation_body = match request.supergraph_request.body().query.clone() {
+            // if the request doesn't have a `query` document, continue with normal execution, which
+            // will result in the normal no-operation error.
+            None => return Ok(request),
+            Some(ob) => ob,
+        };
```

Comment:
nit: I think you can get away with a reference here (and avoid cloning) using [as_ref](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref)

```suggestion
        let operation_body = match request.supergraph_request.body().query.as_ref() {
            // if the request doesn't have a `query` document, continue with normal execution, which
            // will result in the normal no-operation error.
            None => return Ok(request),
            Some(ob) => ob,
        };
```

You would then pass operation_body instead of &operation_body below.

Reply:
Done!